trade-finance-manager-ui/server/controllers/case/cancellation/reason-for-cancelling.controller.get.test.ts

```diff
@@ -92,6 +92,36 @@ describe('getReasonForCancelling', () => {
         ukefDealId,
         dealId,
         reasonForCancelling: reason,
+        previousPage: `/case/${dealId}/deal`,
+      });
+    });
+
+    it('renders the page with the back URL as the check details page when "change" is passed in as a query parameter', async () => {
```

Comment:
Seems that we don't need this test, can just update the test above to consume `getPreviousPageUrlForCancellationFlow` in `previousPage` 😎

Reply:
I'm keen to keep this as is - I like the reassurance of checking on the page level that different results of `getPreviousPageUrlForCancellationFlow` mean different parameters get passed in as part of the view model which requires two tests. 

The alternative to me would be mocking the result of `getPreviousPageUrlForCancellationFlow` and having two test for the two different results of the mocks. But given the preference it seems of minimising mocking, what I've got already here is my next best compromise!

So keen to leave as is if that's ok 👍

Reply:
Really not a fan of this approach as we're testing things that are handled by function B, in function A.

Mocking non-async style helper functions introduces unnecessary test complexity.

As with other PRs / mentioned a few times:

1. The controller simply calls `getPreviousPageUrlForCancellationFlow`.
2. The controller test doesn't need to test anything about `getPreviousPageUrlForCancellationFlow`, just that it's been consumed.
3. With the current approach, if `getPreviousPageUrlForCancellationFlow` changes - the controller test needs to be updated, as well as the `getPreviousPageUrlForCancellationFlow` test - doubling time/effort

The testing approach is inconsistent throughout, including with the current approach in this PR, helper function mockings and the simpler (and IMO more effective, less maintanance) approaches. 

I don't want to block this PR so approving.

Reply:
Appreciate the response, seems like there are some different views around about how best to handle external function calls in unit tests. Given the non-block I'll leave as is here and resolve but maybe one for a wider discussion outside this PR.

Reply:
Hey both,

I agree, it'd be great to have a general chat about this as we seem to be discussing it a lot! 👍  Shall we catch up tomorrow?